mono/mini/debugger-agent.c

```diff
@@ -113,144 +115,6 @@ typedef struct {
 	gboolean setpgid;
 } AgentConfig;
 
-typedef struct
```

Comment:
Why do these struct and define need to move? The state-machine shouldn't have any knowledge of what the various actions are, or you have a leaky abstraction.

Reply:
When the structured logger takes a more enum-based view of the commands and events, a lot of these types will have to be public.

Reply:
Can't the logging be added inside debugger-agent.c ? We are already doing text logging there.

Reply:
These specific named types would have to be made visible in any refactoring of the debugger making any real changes, because they have to be exported in order to export DebuggerTlsData

Reply:
Why do we want the debugger to be a single, impenetrable file with untracked global variables determining state rather than factoring it into easy-to-read subsystems with logged interop points?